dapp/src/features/liquidityPool/components/LiquidityPoolChart/Chart.tsx

```diff
@@ -18,10 +18,20 @@ export function Chart(props: ChartProps) {
   const { data, filteredData, dateRange: initialDateRange, priceRange, tokenPair } = props;
   const { revision } = useSelector(liquidityPoolState);
   const [dateRange, setDateRange] = useState([...initialDateRange] as [string, string]);
-  const traces = generateTraces({ data, filteredData, dateRange, priceRange, tokenPair });
+  const [chartWidth, setChartWidth] = useState(0);
+  const traces = generateTraces({ data, filteredData, dateRange, priceRange, tokenPair, chartWidth });
+  const chartRef = useRef<Plot & { el: HTMLDivElement }>(null);
+
+  const checkChartWidth = useCallback(() => {
```

Comment:
Just wondering about the benefits of useCallback here and `handleInitialized`, while `generateTraces` does not use `useMemo` but does  quite a bit more calculcations.

Reply:
Good catch, added useMemo for generateTraces.